### PR TITLE
Converting azurerm module to az module

### PR DESCRIPTION
**Pull request by ReissJJarvis05**
Converted all code in the guides to use the new Az module instead of the old AzureRm module.

### CODE REVIEW

File: articles/azure/azs-how-move-vm-new-vnet-ps.md [hunk 1]
```diff
@@ -76,86 +76,86 @@ $NewNicName = "<output form="newnic" name="result" style="display: inline;">NewN
...
 if ($VM.StorageProfile.OsDisk.ManagedDisk) {
     $ManagedDisks = $true
-}
-else {
-    $ManagedDisks = $false
-}
+    }
+    else {
+        $ManagedDisks = $false
+  }
```
Comment (wturner123) [inline articles/azure/azs-how-move-vm-new-vnet-ps.md:87]:
```suggestion
}
else {
    $ManagedDisks = $false
}
```

File: articles/azure/azs-how-move-vm-new-vnet-ps.md [hunk 2]
```diff
@@ -76,86 +76,86 @@ $NewNicName = "<output form="newnic" name="result" style="display: inline;">NewN
...
 
 # Create new VM networking resources
-$NewPublicIp = New-AzureRmPublicIpAddress -Name $NewPublicIpName -Location $NewVNet.Location -ResourceGroupName $NewVNet.ResourceGroupName -Sku $PublicIp.Sku.Name -AllocationMethod $PublicIp.PublicIpAllocationMethod
-$NewIpConfig = New-AzureRmNetworkInterfaceIpConfig -Subnet $NewSubnet -Name "ipconfig2" -Primary -PrivateIpAddress $NewPrivateIp -PublicIpAddress $NewPublicIp
-$NewNic = New-AzureRmNetworkInterface -Name $NewNicName -ResourceGroupName $NewVNet.ResourceGroupName -Location $NewVNet.Location -IpConfiguration $NewIpConfig -NetworkSecurityGroupId $Nic.NetworkSecurityGroup.Id -Force
+$NewPublicIp = New-AzPublicIpAddress -Name $NewPublicIpName -Location $NewVNet.Location -ResourceGroupName $NewVNet.ResourceGroupName -Sku $PublicIp.Sku.Name -AllocationMethod $PublicIp.PublicIpAllocationMethod
+$NewIpConfig = New-AzNetworkInterfaceIpConfig -Subnet $NewSubnet -Name "ipconfig2" -Primary -PublicIpAddress $NewPublicIp
+$NewNic = New-AzNetworkInterface -Name $NewNicName -ResourceGroupName $NewVNet.ResourceGroupName -Location $NewVNet.Location -IpConfiguration $NewIpConfig -NetworkSecurityGroupId $Nic.NetworkSecurityGroup.Id -Force
```
Comment (wturner123) [inline articles/azure/azs-how-move-vm-new-vnet-ps.md:103]: What happened with the PrivateIpAddress param?
` -PrivateIpAddress $NewPrivateIp`

**Review by wturner123 — CHANGES_REQUESTED** [review]
Majority of changes are good, but some recurring things across multiple files that need fixing.

File: articles/azure/azs-how-configure-powershell-operators.md [hunk 3]
```diff
@@ -49,16 +49,25 @@ Set-ExecutionPolicy RemoteSigned
...
-Get-Module -Name Azs.*, Azure* -ListAvailable | Uninstall-Module -Force -Verbose
-  
-# Install the AzureRM.BootStrapper module. Select Yes when prompted to install NuGet
-Install-Module -Name AzureRM.BootStrapper -Verbose
+Get-Module -Name Azs.*, Azure*, Az.* -ListAvailable | Uninstall-Module -Force -Verbose
+
+# Install the Az.BootStrapper module
+# On some systems, you may need to run this command to be able to connect to PowerShell Gallery
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+Install-Module -Name Az.BootStrapper -Force -AllowPrerelease
```
Comment (wturner123) [inline articles/azure/azs-how-configure-powershell-operators.md:62]: Think we need to break this up like below. Also reword the comment about the TLS command.
```
# On some older systems, you may need to explicity set the TLS 1.2 security protocol to be able to interact with PowerShell Gallery
[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12

# Install the Az.BootStrapper module
Install-Module -Name Az.BootStrapper -Force -AllowPrerelease
```

File: articles/azure/azs-how-configure-powershell-users.md [hunk 4]
```diff
@@ -49,16 +49,25 @@ Set-ExecutionPolicy RemoteSigned
...
-Get-Module -Name Azs.*, Azure* -ListAvailable | Uninstall-Module -Force -Verbose
-  
-# Install the AzureRM.BootStrapper module. Select Yes when prompted to install NuGet
-Install-Module -Name AzureRM.BootStrapper -Verbose
+Get-Module -Name Azs.*, Azure*, Az.* -ListAvailable | Uninstall-Module -Force -Verbose
+
+# Install the Az.BootStrapper module
+# On some systems, you may need to run this command to be able to connect to PowerShell Gallery
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+Install-Module -Name Az.BootStrapper -Force -AllowPrerelease
```
Comment (wturner123) [inline articles/azure/azs-how-configure-powershell-users.md:62]: Think we need to break this up like below. Also reword the comment about the TLS command.
```
# On some older systems, you may need to explicity set the TLS 1.2 security protocol to be able to interact with PowerShell Gallery
[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12

# Install the Az.BootStrapper module
Install-Module -Name Az.BootStrapper -Force -AllowPrerelease
```

File: articles/azure/azs-how-deploy-sql-template-powershell.md [hunk 5]
```diff
@@ -278,23 +278,23 @@ $SqlServerSKU = "<output form="sqlserversku" name="result" style="display: inlin
...
 # Create a new resource group if it does not exist
 try {
-    $RG = Get-AzureRmResourceGroup -Name $ResourceGroupName -Location $Location -ErrorAction 'SilentlyContinue'
+    $RG = Get-AzResourceGroup -Name $ResourceGroupName -Location $Location -ErrorAction 'SilentlyContinue'
     if (-not $RG) {
         Write-Output -InputObject "Could not find the resource group, creating now..."
-        New-AzureRmResourceGroup -Name $ResourceGroupName -Location $Location -Verbose
-    }
+        New-AzResourceGroup -Name $ResourceGroupName -Location $Location -Verbose
+    }
```
Comment (wturner123) [inline articles/azure/azs-how-deploy-sql-template-powershell.md:297]:
```suggestion
    }
```
Cleaning up white space.

File: articles/azure/azs-how-move-vm-new-vnet-ps.md (hunk 2 above)
Comment (wturner123) [reply articles/azure/azs-how-move-vm-new-vnet-ps.md:103]: @ReissJJarvis05 Was there a specific reason why this was removed?

**Review by wturner123 — CHANGES_REQUESTED** [review]
Just a couple more things to go through.

File: articles/azure/azs-how-configure-powershell-operators.md [hunk 6]
```diff
@@ -49,16 +49,26 @@ Set-ExecutionPolicy RemoteSigned
...
+Install-Module -Name PowerShellGet -Force
+
 # Uninstall existing versions of Azure/Azure Stack Hub PowerShell
-Get-Module -Name Azs.*, Azure* -ListAvailable | Uninstall-Module -Force -Verbose
-  
-# Install the AzureRM.BootStrapper module. Select Yes when prompted to install NuGet
-Install-Module -Name AzureRM.BootStrapper -Verbose
+Get-Module -Name Azs.*, Azure*, Az.* -ListAvailable | Uninstall-Module -Force -Verbose
+
+# On some systems, you may need to run this command to be able to connect to PowerShell Gallery
```
Comment (wturner123) [inline articles/azure/azs-how-configure-powershell-operators.md:59]:
```suggestion
# On some older systems, you may need to explicity set the TLS 1.2 security protocol to be able to interact with PowerShell Gallery
```
Missing the comment change.

File: articles/azure/azs-how-configure-powershell-users.md [hunk 7]
```diff
@@ -49,16 +49,26 @@ Set-ExecutionPolicy RemoteSigned
...
+Install-Module -Name PowerShellGet -Force
+
 # Uninstall existing versions of Azure/Azure Stack Hub PowerShell
-Get-Module -Name Azs.*, Azure* -ListAvailable | Uninstall-Module -Force -Verbose
-  
-# Install the AzureRM.BootStrapper module. Select Yes when prompted to install NuGet
-Install-Module -Name AzureRM.BootStrapper -Verbose
+Get-Module -Name Azs.*, Azure*, Az.* -ListAvailable | Uninstall-Module -Force -Verbose
+
+# On some systems, you may need to run this command to be able to connect to PowerShell Gallery
```
Comment (wturner123) [inline articles/azure/azs-how-configure-powershell-users.md:59]:
```suggestion
# On some older systems, you may need to explicity set the TLS 1.2 security protocol to be able to interact with PowerShell Gallery
```
Missing the comment change.

File: articles/azure/azs-how-configure-powershell-operators.md [hunk 8]
```diff
@@ -49,16 +49,26 @@ Set-ExecutionPolicy RemoteSigned
...
+Install-Module -Name PowerShellGet -Force
+
 # Uninstall existing versions of Azure/Azure Stack Hub PowerShell
-Get-Module -Name Azs.*, Azure* -ListAvailable | Uninstall-Module -Force -Verbose
-  
-# Install the AzureRM.BootStrapper module. Select Yes when prompted to install NuGet
-Install-Module -Name AzureRM.BootStrapper -Verbose
+Get-Module -Name Azs.*, Azure*, Az.* -ListAvailable | Uninstall-Module -Force -Verbose
+
+# On some older systems, you may need to explicity set the TLS 1.2 security protocol to be able to interact with PowerShell Gallery
```
Comment (wturner123) [inline articles/azure/azs-how-configure-powershell-operators.md:59]:
```suggestion
# On some older systems, you may need to explicitly set the TLS 1.2 security protocol to be able to interact with PowerShell Gallery
```
I can't spell...

File: articles/azure/azs-how-configure-powershell-users.md [hunk 9]
```diff
@@ -49,16 +49,26 @@ Set-ExecutionPolicy RemoteSigned
...
+Install-Module -Name PowerShellGet -Force
+
 # Uninstall existing versions of Azure/Azure Stack Hub PowerShell
-Get-Module -Name Azs.*, Azure* -ListAvailable | Uninstall-Module -Force -Verbose
-  
-# Install the AzureRM.BootStrapper module. Select Yes when prompted to install NuGet
-Install-Module -Name AzureRM.BootStrapper -Verbose
+Get-Module -Name Azs.*, Azure*, Az.* -ListAvailable | Uninstall-Module -Force -Verbose
+
+# On some older systems, you may need to explicity set the TLS 1.2 security protocol to be able to interact with PowerShell Gallery
```
Comment (wturner123) [inline articles/azure/azs-how-configure-powershell-users.md:59]:
```suggestion
# On some older systems, you may need to explicitly set the TLS 1.2 security protocol to be able to interact with PowerShell Gallery
```
I can't spell...

File: articles/azure/azs-how-convert-to-managed-disks-ps.md [hunk 10]
```diff
@@ -26,9 +26,6 @@ For more information, see [Introduction to Azure managed disks](https://docs.mic
 
 The following article shows you how to convert a virtual machine from unmanaged to managed disks on UKCloud for Microsoft Azure.
 
-> [!WARNING]
-> At the time of writing, the [ConvertTo-AzureRmVMManagedDisk](https://docs.microsoft.com/en-us/powershell/module/azurerm.compute/convertto-azurermvmmanageddisk?view=azurermps-6.13.0) cmdlet is not supported on Azure Stack Hub and will result in your VM becoming unmanageable. Follow the process below to convert your unmanaged disks safely.
-
```
Comment (wturner123) [inline articles/azure/azs-how-convert-to-managed-disks-ps.md:31]: Need to check whether this is still a valid statement for the Az command.

File: articles/azure/azs-how-configure-custom-script-extension.md [hunk 11]
```diff
@@ -272,42 +272,52 @@ In your PowerShell window:
...
 
 # Add custom script extension to existing Windows VM
 Write-Output -InputObject "Adding custom script extension to VM"
-Set-AzureRmVMCustomScriptExtension -ContainerName $ContainerName -FileName $CustomScriptFileName -Location $Location -Name $CustomScriptFileName -VMName $VMName -ResourceGroupName $RGName -StorageAccountName $CustomScriptStorageAccountName -StorageAccountKey $StorageAccountKey -Run $CommandToExecute -SecureExecution
+$Extensions = Get-AzVMExtensionImage -Location $Location -PublisherName "Microsoft.Azure.Extensions" -Type "CustomScript"
+$ExtensionVersion = $Extensions[0].Version[0..2] -join ""
+$ScriptSettings = @{"fileUris" = @("$ScriptLocation")};
+$ProtectedSettings = @{"storageAccountName" = $CustomScriptStorageAccountName; "storageAccountKey" = $StorageAccountKey; "commandToExecute" = $CommandToExecute};
+Set-AzVMExtension -ResourceGroupName $RGName -Location $Location -VMName $VMName -Name $Extensions[0].Type -Publisher $Extensions[0].PublisherName -ExtensionType $Extensions[0].Type -TypeHandlerVersion $ExtensionVersion -Settings $ScriptSettings -ProtectedSettings $ProtectedSettings
```
Comment (wturner123) [inline articles/azure/azs-how-configure-custom-script-extension.md:320]: Needs looking into, not sure if this is correct.

File: articles/azure/azs-how-configure-custom-script-extension.md [hunk 12]
```diff
@@ -318,25 +328,28 @@ Set-AzureRmVMCustomScriptExtension -ContainerName $ContainerName -FileName $Cust
...
 
-# Add custom script extension to existing Windows VM
-Write-Output -InputObject "Adding custom script extension to VM"
-Set-AzureRmVMCustomScriptExtension -FileUri $FileUri -VMName $VMName -ResourceGroupName $RGName -Name $CustomScriptFileName -Location $Location -Run $CommandToExecute -SecureExecution
+# Add custom script extension to existing Linux VM
+Write-Output -InputObject "Adding custom script extension to existing virtual machine"
+$Extensions = Get-AzVMExtensionImage -Location $Location -PublisherName Microsoft.Azure.Extensions -Type "CustomScript"
+$ExtensionVersion = $Extensions[0].Version[0..2] -join ""
+$ScriptSettings = @{"fileUris" = @($FileUri); "commandToExecute" = $CommandToExecute};
+Set-AzVMExtension -ResourceGroupName $RGName -Location $Location -VMName $VMName -Name $Extensions[0].Type -Publisher $Extensions[0].PublisherName -ExtensionType $Extensions[0].Type -TypeHandlerVersion $ExtensionVersion -Settings $ScriptSettings
```
Comment (wturner123) [inline articles/azure/azs-how-configure-custom-script-extension.md:352]: Needs looking into, not sure if this is correct.

**Review by wturner123 — APPROVED** [review]
LGTM.
Reviewed all articles and double-checked for any remaning -Azure or -AzureRM commands.

**Review by cblackuk — APPROVED** [review]
LGTM

**Review by suehighmoor — APPROVED** [review]
LGTM

**Pull request merged**